### ISSUE TITLE
Sonar Lint does not show all warnings like on SonarQube

### ISSUE THREAD

**Issue by jorenblancquaert** [Riverside-Software/sonar-openedge#1029]
Hi, 

Parsing a file 'HitTester.cls', referencing properties from another class is showing sonar warning on the Sonar Qube server, but not on Sonar Lint in eclipse, Sonar on the fly.

When investigating this 

- Select 1 file 'HitTester.cls' / RMC / SonarLint / Analyze show in 'SonarLint Report':

Unused variable Graph2

> Found 1 issue(s)
> Download server issues for /glims_dev_pro/glims-abl-> sources/be/mips/glims/client/graph/windows/components/cursor/hit/HitTester.cls

- Select 2 files 'HitTester.cls' and 'IAreaDimensions.cls / RMC / SonarLint / Analyze show in 'SonarLint Report':

Conversion from DECIMAL to INT can overflow if source expression is larger than 2^32
Unused variable Graph2
Conversion from DECIMAL to INT can overflow if source expression is larger than 2^32

> Found 3 issue(s)
> Download server issues for /glims_dev_pro/glims-abl-sources/be/mips/glims/client/graph/windows/components/cursor/hit/HitTester.cls`

Both analyses are showing: 
```
Listing file for '[uri=file:/C:/workspace/OE12GL/glims_dev_pro/glims-abl-sources/be/mips/glims/client/graph/windows/components/cursor/hit/HitTester.cls]' not found or contains space character - > > Was looking for 'null'
No TypeInfo found for Graph
No TypeInfo found for e
No TypeInfo found for e
No TypeInfo found for HitDataPoint
No TypeInfo found for HitDataPoint
No TypeInfo found for AreaDimensions
No TypeInfo found for AreaDimensions
No TypeInfo found for e
No TypeInfo found for e
No TypeInfo found for ResetRequested
No TypeInfo found for OpenEdge.Core.EventArgs
No TypeInfo found for PointHit
No TypeInfo found for UnhandledHit
10 milliseconds to generate ParseUnit
```

The simplified lines with issues only seen when both files are analyzed togehter: 
```
METHOD PUBLIC VOID HitTest(AreaDimensions AS IAreaDimensions, e AS MouseEventArgs):
DEFINE VARIABLE XPointPixel AS INTEGER NO-UNDO.
DEFINE VARIABLE YPointPixel AS INTEGER NO-UNDO.
XPointPixel = AreaDimensions:XValueToPixel(XValue).
YPointPixel = AreaDimensions:YValueToPixel(YValue).
END METHOD.
```

Can we use SonarLint On the Fly on 1 sources to show also the warning about overflow (or others) related to another interface/class ?

The verbose logging is attached to this log.
[ANL_Verbose_SingleFile.txt](https://github.com/Riverside-Software/sonar-openedge/files/10861726/ANL_Verbose_SingleFile.txt)
[ANL_Verbose_TwoFiles.txt](https://github.com/Riverside-Software/sonar-openedge/files/10861727/ANL_Verbose_TwoFiles.txt)

Thanks
Joren
Clinisys / MIPS

**Comment by jorenblancquaert** [comment]
Hi Gilles,
Attached the screenshots.

So is it possible to also see the 2 others warnings on HitTester.cls with the Sonar Lint on the fly ?
Now it only pops up when SonarQube server is analyzing the issues, and not directly to each developer inside eclipse.

![2files_Problems](https://user-images.githubusercontent.com/20811052/223067174-76f81cb7-234e-4cbd-bca0-90ee34da1b56.png)
![1file_OnTheFly](https://user-images.githubusercontent.com/20811052/223067179-fae36a46-dc99-4314-9957-0b5a8661d943.png)

**Comment by gquerret** [comment]
If you check "Run SonarLint automatically", SonarLint executes the analysis when the file is saved.
![image](https://user-images.githubusercontent.com/6498723/223485575-2bbf2bce-e046-40a4-9f87-dbb865eb8771.png)

Otherwise, you have to right-click on the file, and execute the analysis. You can also use the annotations.

The SonarLint on the fly view only shows issues found in the currently opened file (which has the focus).

**Comment by gquerret** [comment]
https://community.sonarsource.com/t/sonar-lint-on-the-fly-view-is-not-respecting-the-filter-to-display-all-violations/68561/3?u=gquerret

**Comment by dominickdm** [comment]
I'm not sure the issue is clear.

Here is another example:

If you just do a sonar scan on one file (or just save the file) => no issues are found
(scan on FamilyImporter)
![image](https://user-images.githubusercontent.com/40289256/233343034-2247e389-43e2-45f4-a7a0-7b4dc49993f6.png)

If you do a sonarlint on the same file + the involved class (FamilyPedigreeBL), you get 7 issues (that are also reported by the server scan).

![image](https://user-images.githubusercontent.com/40289256/233341599-f64f1b49-27cf-45e9-85c7-45cd81bfd747.png)

This doesn't seem normal to me? Is FamilyImporter should be aware of the return types of the FamilyPedigreeBL, even if not explicitly scanned, no?

**Comment by gquerret** [comment]
@dominickdm I have a similar behavior in my test environment. That will be investigated.

**Comment by gquerret** [comment]
Fixed in [SonarLint 9.1.2906](https://github.com/Riverside-Software/sonar-openedge/releases/tag/V2.24.1)
